docs/http-api-intigration.md

```diff
@@ -0,0 +1,65 @@
+
+## Helping out RESTful with JSON API or GraphQL
+
+### Design
+The design of RESTful API is to return resources using a Univerisal Resource Identifier. A client could make a requests such as `GET https//example.com/book/Hobbit`. This works in the simplest way to return all data from that resource even if all the fields aren't needed. If you needed specific information like `author` and `title`, you'd need to send multiple request: `GET /Hobbit/author`, `GET /Hobbit/title` This causes a lot of bandwidth between the client and server.
```

Comment:
One of the core constraints* RESTful API  [.... about the URI]. However, there is not an explicit constraint for limiting or including additional fields in a resource response (e.g. techniques "Sparse Fields" and "Compound Documents" and "Pagination"). This leaves it up to the implementors to decide how or even **if** to implement these features. 

This is where [JSON:API](https://jsonapi.org/format/) and GraphQL (among a few others) provide clear constraints and features for the previously listed omissions RESTful API. 

While there are many features and constraints that JSON:API and GraphQL both provide, we will focus on the ones for reducing the size of resource responses and the number of requests need to get the data for a given application state. 

\* _According to _

Reply:
Your wording is so good that I want to use most of it in the final product. Is that fine if I copy it?